common/src/claims.rs

```diff
@@ -182,11 +234,45 @@ pub struct Claim {
     pub scopes: Vec<Scope>,
     /// The original token that was parsed
     pub(crate) token: Option<String>,
+    /// A struct that holds the account limits.
+    pub limits: Limits,
+    /// The account tier of the subject.
+    pub tier: AccountTier,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+pub struct Limits {
+    /// The amount of projects this user can create.
+    project_limit: u32,
+}
+
+impl Default for Limits {
+    fn default() -> Self {
+        Self {
+            project_limit: MAX_PROJECTS_DEFAULT,
+        }
+    }
+}
+
+impl Limits {
+    pub fn new(project_limit: u32) -> Self {
+        Self { project_limit }
+    }
+}
+
+pub trait ClaimExt {
+    fn project_limit(&self) -> u32;
```

Comment:
It would be better to have this method as `can_create_project()` to prevent the logic from spilling into the handlers (controllers).

Reply:
It would also be good to put this trait in a separate file.

Reply:
> It would also be good to put this trait in a separate file.

Why?

> can_create_project

Are you saying that the business logic should live in that function? Then it has to take in `current_project_count` or something.

Reply:
I implemented these changes last night: [f900d1a](https://github.com/shuttle-hq/shuttle/pull/1382/commits/f900d1a379e792e7bf50eaafdf5b354869650c94)

It feels more organized, but it meant needing to make some changes to the test code in gateway.

Reply:
@jonaro00 to have the business logic in one file. Thereby being easier to update :smile: